src/elasticsearch/esSearch.ts

```diff
@@ -0,0 +1,104 @@
+import { Ctx } from '../processor';
+import { SubsocialElasticApi } from '@subsocial/elasticsearch';
+import {
+  buildElasticSearchQuery,
+  getElasticQueryParamsDecorated
+} from './searchUtils';
+import {
+  ElasticQueryParamsWithSpaceIdRaw,
+  ESErrorType,
+  ESQueryResponseContent,
+  ESSearchResultHitItem,
+  OkOrError
+} from './types';
+import { ErrorResponseBase } from '@elastic/elasticsearch/lib/api/types';
+import { getChain } from '../chains';
+import { parseNumStr } from '@subsocial/utils';
+const chainConfig = getChain();
+const maxResultLimit: number =
+  parseNumStr(chainConfig.config.elasticSearchMaxResultLimit) || 20;
+
```

Comment:
Looks like you have same code for instance creating, so maybe can you extract it to parent class and use for both cildren?